src/core_codemods/defectdojo/results.py

```diff
@@ -0,0 +1,65 @@
+import json
+from functools import cache
+from pathlib import Path
+
+import libcst as cst
+from libcst._position import CodeRange
+from typing_extensions import Self, override
+
+from codemodder.context import CodemodExecutionContext
+from codemodder.result import LineInfo, Location, Result, ResultSet
+
+
+class DefectDojoLocation(Location):
+    @classmethod
+    def from_finding(cls, finding: dict) -> Self:
+        return cls(
+            file=Path(finding["file_path"]),
+            # TODO: parse snippet from "description" field?
+            start=LineInfo(finding["line"]),
+            end=LineInfo(finding["line"]),
+        )
+
+
+class DefectDojoResult(Result):
+    @classmethod
+    def from_finding(cls, finding: dict) -> Self:
+        return cls(
+            rule_id=finding["title"],
+            locations=[DefectDojoLocation.from_finding(finding)],
+        )
+
+    @override
+    def match_location(self, pos: CodeRange, node: cst.CSTNode) -> bool:
+        """
+        Match location for DefectDojo results
+
+        Since DefectDojo does not provide column information, we can only match based on line number.
+        We check whether the start line of the result is within the range of the node.
+        """
+        del node
+        return any(
+            pos.start.line <= location.start.line <= pos.end.line
+            for location in self.locations
```

Comment:
could you reuse the existing `fuzzy_column_match` function here?

Reply:
I'm looking at that function and it just applies to columns, but now I'm questioning whether the logic there is correct. In either case, DefectDojo doesn't give us column information so that function can't help us.

Reply:
@andrecsilva recently commented something similar that leads to the general question of: why do we use column info in the first place? Perhaps there are strong reasons in some cases with lines that have multiple nodes. we should explore this later on

Reply:
We can use column information to differentiate ambiguous cases. For example, imagine we get a finding that says there's a SQL injection vulnerability on this line:
```python
result = cursor.execute(query).something_else(whatever)
```

Without column information, we don't know which of those call sites potentially need to get modified and if we're not careful we would end up modifying both of them.